### PR TITLE
Make Rx family covariant

### LINKED ISSUE #1609: Make ObservableK and the family covariant
We found that when working with Observable and Either, mapping into `Either.left` would cause an error because the other side can't be hinted and Observable is invariant, so you need to create a local variable to give the correct type to Nothing.

We can fix this in ObservableK and suppress the variance issues.

cc @Laimiux @lenguyenthanh

### PR DESCRIPTION
**Pull request by lenguyenthanh**
Here is my first try to fix #1609.
I only made `ObservableK` covariant this time, after making sure everything fine with `ObservableK` I'll continue to work on other types of Rx family.

P/S: I had to remove `ObservableKKindedJ<A>` from inherited list because `io.kindedj.Hk<ForObservableK, A>` is invariant.

### CODE REVIEW

File: modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt [hunk 1]
```diff
@@ -23,18 +23,18 @@ class ForObservableK private constructor() {
...
 @Suppress("UNCHECKED_CAST", "NOTHING_TO_INLINE")
 inline fun <A> ObservableKOf<A>.fix(): ObservableK<A> =
   this as ObservableK<A>
 
 fun <A> Observable<A>.k(): ObservableK<A> = ObservableK(this)
 
+@Suppress("UNCHECKED_CAST")
 fun <A> ObservableKOf<A>.value(): Observable<A> =
-  fix().observable
+  fix().observable as Observable<A>
```
Comment (pakoito) [inline modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt]: Why is this necesssary? What's the complaint?

**Comment by pakoito** [comment]
So far so good

File: modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt (hunk 1 above)
Comment (lenguyenthanh) [reply modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt]: Screenshot 2019-09-24 at 22 23 48
It seems it cannot auto cast `Observable<out A>` to `Observable<A>`

Comment (pakoito) [reply modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt]: see if there's any of the @annotations that may fix it :D

Comment (lenguyenthanh) [reply modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt]: where should I look for annotations? I checked here: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-annotation.html, but I didn't find any thing useful.

Comment (pakoito) [reply modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt]: JvmSuppressWildcards or JvmWildcard

Comment (lenguyenthanh) [reply modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt]: Both of them dont' work :(

Comment (lenguyenthanh) [reply modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt]: do I need to do anything to fix this? @pakoito

File: modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/FlowableK.kt [hunk 2]
```diff
@@ -24,17 +24,17 @@ class ForFlowableK private constructor() {
   companion object
 }
 typealias FlowableKOf<A> = arrow.Kind<ForFlowableK, A>
-typealias FlowableKKindedJ<A> = io.kindedj.Hk<ForFlowableK, A>
```
Comment (JorgeCastilloPrz) [inline modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/FlowableK.kt]: Are we removing KindedJ support? @pakoito

Comment (pakoito) [reply modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/FlowableK.kt]: No, covariance needs a compatibility function that's on the `arrow-kindedj` module.

File: modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt (hunk 1 above)
Comment (pakoito) [reply modules/fx/arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/ObservableK.kt]: No, leave it as is.

I'd suggest testing a bit just in case.

**Comment by lenguyenthanh** [comment]
Any thing I can do to make this go forward?

**Comment by pakoito** [comment]
I'll click the green button as soon as all the checks pass, and set a reminder for it!

**Comment by lenguyenthanh** [comment]
No hurry, I'm just checking ;)

**Comment by pakoito** [comment]
Neat :D If you wanna round it up, try the same with Project Reactor!

**Comment by lenguyenthanh** [comment]
Will do it soon!

**Pull request merged**